pkg/kubelet/cm/container_manager.go

```diff
@@ -115,10 +116,10 @@ type ContainerManager interface {
 	// GetPodCgroupRoot returns the cgroup which contains all pods.
 	GetPodCgroupRoot() string
 
-	// GetPluginRegistrationHandler returns a plugin registration handler
+	// GetPluginRegistrationHandlers returns a plugin registration handlers
 	// The pluginwatcher's Handlers allow to have a single module for handling
 	// registration.
-	GetPluginRegistrationHandler() cache.PluginHandler
+	GetPluginRegistrationHandlers(kubeClient kubernetes.Interface, getNode func() (*v1.Node, error)) map[string]cache.PluginHandler
```

Comment:
I'm conceptually fine unifying the flows but what irks me is the creation API is so wildly different among the two.

Reply:
Question: linux container manager has a kubernetes client (see container_manager_linux.go::NewContainerManager); windows doesn't support DRA, not sure when that support will come and most likely it will need a kubernetes client also at that point. So, can we reuse the container manager kube client? I'm trying to simplify the interface here because really feels disjoint

Reply:
yes, good idea. I am also thinking if I should do the same for `getNode` and just pass it into the container manager constructor instead of here. wdyt?